src/algorithms/particle/CaloRemnantCombiner.h

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2025 Subhadip Pal
+
+#pragma once
+
+#include <algorithms/algorithm.h>
+#include <edm4eic/ClusterCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+#include <string>
+#include <string_view>
+
+#include "CaloRemnantCombinerConfig.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+using CaloRemnantCombinerAlgorithm =
+    algorithms::Algorithm<algorithms::Input<std::vector<edm4eic::ClusterCollection>>,
+                          algorithms::Output<edm4eic::ReconstructedParticleCollection>>;
+
+class CaloRemnantCombiner : public CaloRemnantCombinerAlgorithm,
```

Comment:
I'd suggest adding docstrings in several places. These should get propagated to our doxygen page, and help explain what's happening in the code for anyone who's looking at the source code!
```suggestion
// ==========================================================================
//! Calorimeter Remnant Cluster Combiner
// ==========================================================================
/*! An algorithm which takes multiple calorimeter cluster collections and combines them into
 *  neutral-particle candidates based on distance matching.
 */
class CaloRemnantCombiner : public CaloRemnantCombinerAlgorithm,
```